src/cobra/io/mat.py

```diff
@@ -284,139 +673,148 @@ def from_mat_struct(
+            ]
+        # Other matlab files seem normal.
+        else:
+            rxn_subsystems = _cell_to_str_list(m["subSystems"][0, 0])
+    except (IndexError, ValueError):
+        # TODO: use custom cobra exception to handle exception
+        pass
+    for i in range(len(rxn_ids)):
+        new_reaction = Reaction(
+            id=rxn_ids[i], lower_bound=rxn_lbs[i], upper_bound=rxn_ubs[i]
+        )
+        if rxn_names:
+            new_reaction.name = rxn_names[i]
+        if rxn_subsystems:
+            new_reaction.subsystem = rxn_subsystems[i]
+        if rxn_gene_rules:
+            new_reaction.gene_reaction_rule = rxn_gene_rules[i]
         new_reactions.append(new_reaction)
-    model.add_reactions(new_reactions)
-    set_objective(model, coefficients)
-
-    coo = scipy_sparse.coo_matrix(m["S"][0, 0])
-    for i, j, v in zip(coo.row, coo.col, coo.data):
-        model.reactions[j].add_metabolites({model.metabolites[i]: v})
-    return model
-
-
-def _check(result: Dict[str, str]) -> None:
-    """Ensure success of a `pymatbridge` operation.
-
-    Parameters
-    ----------
-    result : dict
-        The dictionary obtained from `pymatbridge` with keys as 'message',
-        'result', and 'success'.
-
-    Raises
-    ------
-    RuntimeError
-        If setting variable via `pymatbridge` fails.
-
-    """
-    if result["success"] is not True:
-        # TODO: verify if key 'content' is valid as docs don't state about it
-        raise RuntimeError(result["content"]["stdout"])
+    mat_parse_annotations(new_reactions, m, _d_replace=D_REACTION)
+    mat_parse_notes(new_reactions, m, _d_replace=D_REACTION_NOTES)
 
+    csc = scipy_sparse.csc_matrix(m["S"][0, 0])
+    for i in range(csc.shape[1]):
+        stoic_dict = {
+            model.metabolites[j]: csc[j, i] for j in csc.getcol(i).nonzero()[0]
+        }
+        new_reactions[i].add_metabolites(stoic_dict)
 
-def model_to_pymatbridge(
-    model: Model,
-    variable_name: str = "model",
-    matlab: Optional["pymatbridge.Matlab"] = None,
-) -> None:
-    """Send the model to a MATLAB workspace through `pymatbridge`.
-
-    This model can then be manipulated through the cobratoolbox.
+    model.add_reactions(new_reactions)
 
-    Parameters
-    ----------
-    model : cobra.Model
-        The model to send to MATLAB workspace.
-    variable_name : str, optional
-        The variable name to which the model will be assigned in the
-        MATLAB workspace (default 'model').
-    matlab : pymatbridge.Matlab, optional
-        The MATLAB workspace to which the variable will be sent. If None,
-        the variable will be sent to the same environment used in IPython
-        magics.
+    if "c" in m.dtype.names:
+        c_vec = _cell_to_float_list(m["c"][0, 0])
+        coefficients = dict(zip(new_reactions, c_vec))
+        set_objective(model, coefficients)
+    else:
+        warn("Objective vector `c` not found.")
```

Comment:
I would transfer all wanrings to logger warnings since those can be manager better and play nice with multiprocessing (thread safe).

Reply:
Done